filament/src/components/RenderableManager.cpp

```diff
@@ -202,6 +203,88 @@ RenderableManager::Builder& RenderableManager::Builder::enableSkinningBuffers(bo
+        utils::FixedCapacityVector<float2> vertexData(pairsPerVertexCount);
+        std::memcpy(&vertexData[0], &indicesAndWeights[offset + iVertex * pairsPerVertexCount],
+             pairsPerVertexCount * sizeof(float2));
+        bonePairs[iVertex] = vertexData;
+    }
+  return boneIndicesAndWeights(primitiveIndex, bonePairs);
+}
+
+RenderableManager::Builder& RenderableManager::Builder::boneIndicesAndWeights(
+             math::float2 const* indicesAndWeightsArray, size_t count, size_t offset){
+    std::vector<Entry>& entries = mImpl->mEntries;
+    size_t primitiveCount = entries.size();
+    size_t allVerticesCount = 0;
+    // number off all vertices of all primitives calculation
+    for (size_t iEntry = 0; iEntry < primitiveCount; iEntry++){
+        allVerticesCount += entries[iEntry].vertices->getVertexCount();
+    }
+    size_t pairsPerAllVerticesCount = count / allVerticesCount;
+    ASSERT_PRECONDITION(pairsPerAllVerticesCount > 0 && count % allVerticesCount == 0,
+        "bone indices and weights pairs count (%u) must be a multiple of vertex counts of all primitives (%u)",
+        count, allVerticesCount);
+    size_t pairSkipped = offset;
+    for (size_t iEntry = 0; iEntry < primitiveCount; iEntry++){
+        size_t  pairsPerPrimitiveCount = pairsPerAllVerticesCount * entries[iEntry].vertices->getVertexCount();
+        boneIndicesAndWeights(iEntry,
+             indicesAndWeightsArray + pairSkipped,
+             pairsPerPrimitiveCount);
+        pairSkipped += pairsPerAllVerticesCount;
+    }
+    return *this;
+}
+
+RenderableManager::Builder& RenderableManager::Builder::boneIndicesAndWeights(size_t primitiveIndex,
+               const  utils::FixedCapacityVector< utils::FixedCapacityVector<filament::math::float2>> &indicesAndWeightsVector) {
+    auto count = indicesAndWeightsVector.size();
+    if (!count) {
+      // skip if no skinning data
+      //mImpl->mEntries[primitiveIndex].skinning.bonePairsSet = true;
+      return *this;
+    }
+    std::vector<Entry>& entries = mImpl->mEntries;
+    ASSERT_PRECONDITION(primitiveIndex < entries.size() && primitiveIndex >= 0,
+        "[primitive @ %u] index is out of size (%u)", primitiveIndex, entries.size());
+    auto vertexCount = entries[primitiveIndex].vertices->getVertexCount();
+    ASSERT_PRECONDITION(vertexCount > 0,
+        "[primitive @ %u] no defined vertices (%u)", primitiveIndex, vertexCount);
+    auto pairsPerVertexCount = count / vertexCount;
+    ASSERT_PRECONDITION(pairsPerVertexCount > 0 && count % vertexCount == 0,
+        "[primitive @ %u] bone indices and weights pairs count (%u) must be a multiple of vertex count (%u)",
+        primitiveIndex, count, vertexCount);
+    auto& skinning = entries[primitiveIndex].skinning;
+    ASSERT_PRECONDITION(skinning.bonePairs.empty(),
+        "[primitive @ %u] bone indices and weights data already exists", primitiveIndex);
+    skinning.bonePairs = indicesAndWeightsVector; //copy
```

Comment:
With a `&&` override you could do a `std::move` here, I think it can be worth it.